calour/dsfdr.py

```diff
@@ -153,6 +156,10 @@ def dsfdr(data, labels, transform_type='rankdata', method='meandiff',
 
     logger.debug('dsfdr using fdr method: %s' % fdr_method)
 
+    # if random_state is supplied, create a numpy RandomState
+    # (if random seed is None, use the numpy seed)
+    random_state = np.random.RandomState(random_state)
```

Comment:
this will fail if `random_state` is already a RandomState instance? shall we move the random_state code to a util function? like sklearn.utils.check_random_state()